hack/jenkins/test-flake-chart/report_flakes/report_flake.go

```diff
@@ -0,0 +1,258 @@
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/csv"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"sort"
+	"strconv"
+	"strings"
+
+	"cloud.google.com/go/storage"
+)
+
+const unknown float64 = -1.0
+
+type shortSummary struct {
+	NumberOfTests int                `json:"NumberOfTests"`
+	NumberOfFail  int                `json:"NumberOfFail"`
+	NumberOfPass  int                `json:"NumberOfPass"`
+	NumberOfSkip  int                `json:"NumberOfSkip"`
+	FailedTests   []string           `json:"FailedTests"`
+	PassedTests   []string           `json:"PassedTests"`
+	SkippedTests  []string           `json:"SkippedTests"`
+	Durations     map[string]float64 `json:"Durations"`
+	TotalDuration float64            `json:"TotalDuration"`
+	GopoghVersion string             `json:"GopoghVersion"`
+	GopoghBuild   string             `json:"GopoghBuild"`
+	Detail        struct {
+		Name     string `json:"Name"`
+		Details  string `json:"Details"`
+		PR       string `json:"PR"`
+		RepoName string `json:"RepoName"`
+	} `json:"Detail"`
+}
+
+// parseEnvironmentList reads the existing environments from the file
+func parseEnvironmentList(listFile string) ([]string, error) {
+	data, err := os.ReadFile(listFile)
+	if err != nil {
+		return nil, err
+	}
+	return strings.Split(strings.TrimSpace(string(data)), "\n"), nil
+}
+
+func testSummariesFromGCP(ctx context.Context, pr, rootJob string, envList []string, client *storage.Client) (map[string]*shortSummary, error) {
+	envToSummaries := map[string]*shortSummary{}
+	for _, env := range envList {
+		summary, err := testSummaryFromGCP(ctx, pr, rootJob, env, client)
+		if err != nil {
+			return nil, fmt.Errorf("failed to fetch %s test summary from gcp, err: %v", env, err)
+		}
+		if summary != nil {
+			// if the summary is nil(missing) we just skip it
+			envToSummaries[env] = summary
+		}
+	}
+	return envToSummaries, nil
+}
+
+// testSummaryFromGCP gets the summary of a test for the specified env.
+func testSummaryFromGCP(ctx context.Context, pr, rootJob, env string, client *storage.Client) (*shortSummary, error) {
+
+	btk := client.Bucket("minikube-builds")
+	obj := btk.Object(fmt.Sprintf("logs/%s/%s/%s_summary.json", pr, rootJob, env))
+
+	reader, err := obj.NewReader(ctx)
+	if errors.Is(err, storage.ErrObjectNotExist) {
+		// if this file does not exist, just skip it
+		return nil, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	// read the file
+	data, err := io.ReadAll(reader)
+	if err != nil {
+		return nil, err
+	}
+
+	var summary shortSummary
+	if err = json.Unmarshal(data, &summary); err != nil {
+		return nil, fmt.Errorf("failed to deserialize the file: %v", err)
+	}
+	return &summary, nil
+
+}
+
+// flakeRate downloads recent flake rates from GCS, and returns a map{env->map{testname->flake rate}}
```

Comment:
```suggestion
// flakeRate downloads recent flake rates from GCS, and returns a
// map{env->map{testname->flake rate}}
```
Shortening long comments for better readability